### ISSUE TITLE
False-positive "Use of moved value"

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#3970]
```rust
struct S;
fn main() {
    let mut a = Some(S);
    while let Some(b) = a {
        match false {
            true => { a = Some(S) }
            _ => break
        }
    }
}
```
